packages/zoe/src/contractFacet/zcfSeat.js

```diff
@@ -334,32 +240,158 @@ export const createSeatManager = (
       hasStagedAllocation: ({ self }) => hasStagedAllocation(self),
     },
   );
-  const makeZCFSeat = ({ proposal, initialAllocation, seatHandle }) => {
-    const zcfSeat = makeZCFSeatInternal(proposal);
-    activeZCFSeats.init(zcfSeat, initialAllocation);
-    zcfSeatToSeatHandle.init(zcfSeat, seatHandle);
-    return zcfSeat;
-  };
 
   const replaceDurableWeakMapStore = (baggage, key) => {
     const mapStore = makeScalarBigWeakMapStore(key, { durable: true });
     baggage.set(key, mapStore);
     return mapStore;
   };
 
-  /** @type {DropAllReferences} */
-  const dropAllReferences = () => {
-    activeZCFSeats = replaceDurableWeakMapStore(zcfBaggage, 'activeZCFSeats');
-    zcfSeatToSeatHandle = replaceDurableWeakMapStore(
-      zcfBaggage,
-      'zcfSeatToSeatHandle',
-    );
+  const ZcfSeatManagerGuard = {
+    seatManager: M.interface('ZcfSeatManager', {
+      makeZCFSeat: M.call(
+        M.partial({
+          proposal: ProposalShape,
+          initialAllocation: AmountKeywordRecordShape,
+          seatHandle: SeatShape,
+          offerArgs: M.any(),
+        }),
+      ).returns(M.remotable('zcfSeat')),
+      reallocate: M.call(M.remotable('zcfSeat'), M.remotable('zcfSeat'))
+        .rest(M.arrayOf(M.remotable('zcfSeat')))
+        .returns(),
+      dropAllReferences: M.call().returns(),
+    }),
+    zcfMintReallocator: M.interface('MintReallocator', {
+      reallocate: M.call(SeatShape, AmountKeywordRecordShape).returns(),
+    }),
   };
 
-  return harden({
-    makeZCFSeat,
-    reallocate,
-    reallocateForZCFMint,
-    dropAllReferences,
-  });
+  const makeSeatManagerKit = vivifyFarClassKit(
+    zcfBaggage,
+    'ZcfSeatManager',
+    ZcfSeatManagerGuard,
+    initEmpty,
+    {
+      seatManager: {
+        makeZCFSeat({ proposal, initialAllocation, seatHandle }) {
+          const zcfSeat = makeZCFSeatInternal(proposal);
+          activeZCFSeats.init(zcfSeat, initialAllocation);
+          zcfSeatToSeatHandle.init(zcfSeat, seatHandle);
+          return zcfSeat;
+        },
+
+        reallocate(/** @type {ZCFSeat[]} */ ...seats) {
+          seats.forEach(assertActive);
+          seats.forEach(assertStagedAllocation);
+
+          // Ensure that rights are conserved overall.
+          const flattenAllocations = allocations =>
+            allocations.flatMap(Object.values);
+
+          const previousAllocations = seats.map(seat =>
+            seat.getCurrentAllocation(),
+          );
+          const previousAmounts = flattenAllocations(previousAllocations);
+
+          const newAllocations = seats.map(seat => seat.getStagedAllocation());
+          const newAmounts = flattenAllocations(newAllocations);
+
+          assertRightsConserved(previousAmounts, newAmounts);
```

Comment:
Would this read better as a single block with no medial blank lines?
```suggestion
          // Ensure that rights are conserved overall.
          const flattenAllocations = allocations =>
            allocations.flatMap(Object.values);
          const previousAllocations = seats.map(seat =>
            seat.getCurrentAllocation(),
          );
          const previousAmounts = flattenAllocations(previousAllocations);
          const newAllocations = seats.map(seat => seat.getStagedAllocation());
          const newAmounts = flattenAllocations(newAllocations);
          assertRightsConserved(previousAmounts, newAmounts);
```

Reply:
Looks done enough. Resolving.